contrib/bsp/src/mill/contrib/bsp/ModuleUtils.scala

```diff
@@ -10,247 +11,174 @@ import mill.scalajslib.ScalaJSModule
+
+    target.setData(
+      new ScalaBuildTarget(
+        "org.scala-lang",
+        BuildInfo.scalaVersion,
+        Util.scalaBinaryVersion(BuildInfo.scalaVersion),
+        ScalaPlatform.JVM,
+        classpath.map(_.path.toNIO.toUri.toString).asJava
+      )
+    )
+
+    target
   }
 
   /**
-    * Compute the BuildTarget associated with the given mill
-    * JavaModule, which is any module present in the working
-    * directory, but it's not the root module itself.
-    *
+   * Compute the BuildTarget associated with the given mill
+   * JavaModule, which is any module present in the working
+   * directory, but it's not the root module itself.
+   *
     * @param module      any in-project mill module
-    * @param evaluator   mill evaluator
-    * @param moduleIdMap mapping from each mill JavaModule
-    *                    contained in the working directory and
-    *                    a BuildTargetIdentifier associated
-    *                    with it.
-    * @return inner BuildTarget
-    */
-  def getRegularTarget(
-                        module: JavaModule,
-                        evaluator: Evaluator,
-                        moduleIdMap: Map[JavaModule, BuildTargetIdentifier]): BuildTarget = {
+   * @param evaluator   mill evaluator
+   * @return inner BuildTarget
+   */
+  def getTarget(module: JavaModule, evaluator: Evaluator): BuildTarget = {
     val dataBuildTarget = computeBuildTargetData(module, evaluator)
-    val capabilities = getModuleCapabilities(module, evaluator)
-    val buildTargetTag: List[String] = module match {
-      case m: TestModule => List(BuildTargetTag.TEST)
-      case m: JavaModule => List(BuildTargetTag.LIBRARY, BuildTargetTag.APPLICATION)
-    }
-
-    val dependencies = module match {
-      case m: JavaModule => m.moduleDeps.map(dep => moduleIdMap(dep)).toList.asJava
+    val capabilities = getModuleCapabilities(module)
+    val buildTargetTag = module match {
+      case _: TestModule => Seq(BuildTargetTag.TEST)
+      case _: JavaModule => Seq(BuildTargetTag.LIBRARY, BuildTargetTag.APPLICATION)
     }
 
-    val buildTarget = new BuildTarget(moduleIdMap(module),
-                                      buildTargetTag.asJava,
-                                      List("scala", "java").asJava,
-                                      dependencies,
-                                      capabilities)
+    val buildTarget = new BuildTarget(
+      getTargetId(module),
+      buildTargetTag.asJava,
+      Seq("scala", "java").asJava,
+      module.moduleDeps.map(getTargetId).toList.asJava,
+      capabilities
+    )
     if (module.isInstanceOf[ScalaModule]) {
       buildTarget.setDataKind(BuildTargetDataKind.SCALA)
     }
     buildTarget.setData(dataBuildTarget)
-    buildTarget.setDisplayName(moduleName(module.millModuleSegments))
-    buildTarget.setBaseDirectory(module.intellijModulePath.toIO.toURI.toString)
+    buildTarget.setDisplayName(module.millModuleSegments.render)
```

Comment:
I removed `moduleName` for `.render`. That makes display of modules in Idea (and other BSP clients) look exactly the same as mill cli.